lib/subscription-connection.js

```diff
@@ -171,73 +171,98 @@ module.exports = class SubscriptionConnection {
       }
     }
 
-    const sc = new SubscriptionContext({
-      fastify: this.fastify,
-      pubsub: this.subscriber
-    })
-
-    const schema = this.fastify ? this.fastify.graphql.schema : undefined
-
-    // Trigger preSubscriptionParsing hook
-    if (this.context.preSubscriptionParsing !== null && typeof schema !== 'undefined' && typeof query === 'string') {
-      await preSubscriptionParsingHandler({ schema, source: query, context })
-    }
     const document = typeof query !== 'string' ? query : parse(query)
-    this.subscriptionContexts.set(id, sc)
 
-    // Trigger preSubscriptionExecution hook
-    if (this.context.preSubscriptionExecution !== null && typeof schema !== 'undefined') {
-      await preSubscriptionExecutionHandler({ schema, document, context })
+    if (!document) {
+      throw new Error('Must provide document.')
     }
 
-    let subscriptionLoaders
-    if (this.fastify && this.fastify[kSubscriptionFactory]) {
-      subscriptionLoaders = this.fastify[kSubscriptionFactory]
-    }
+    const operationAST = getOperationAST(document, operationName)
+
+    if (operationAST.operation === 'subscription') {
+      // execute Subscription
```

Comment:
Can you extract the content of this if in its own function? This function is becoming unreadable.